src/mscorlib/src/System/ArraySegment.cs

```diff
@@ -145,6 +167,34 @@ public bool Equals(ArraySegment<T> obj)
             return obj._array == _array && obj._offset == _offset && obj._count == _count;
         }
 
+        public ArraySegment<T> Slice(int index)
```

Comment:
C# team is considering adding slicing syntax. It's not yet clear what would libraries have to do to support it. It could be Slice(int) method (just like here) or Slice(Range) method. 

When this C# syntax ships, it would be great if ArraySegment supported it. 

@jaredpar, @MadsTorgersen, any word on the syntax? Shoudl we just assume the Range solution? Should we add Range struct to the platform?